backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flare/internal/FlareShape.kt

```diff
@@ -0,0 +1,111 @@
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.flare.BpkFlarePointerDirection
+import net.skyscanner.backpack.compose.flare.BpkFlareRadius
+import net.skyscanner.backpack.compose.tokens.BpkDimension
+
+internal class FlareShape(
+  private val radius: BpkFlareRadius,
+  private val pointerDirection: BpkFlarePointerDirection,
+) : Shape {
+
+  override fun createOutline(size: Size, layoutDirection: LayoutDirection, density: Density): Outline {
+    return Outline.Generic(
+      path = Path().apply {
+        val flareHeight = with(density) { FlareHeight.toPx() }
+        var scale = flareHeight / FlareVectorHeight
+        if (pointerDirection == BpkFlarePointerDirection.Up) {
+          scale *= -1
+        }
+        addRect(flareHeight, size, density, radius, pointerDirection)
+        addFlare(flareHeight, scale, size, pointerDirection)
+      }
+    )
+  }
+}
+
+private fun Path.addRect(
+  flareHeight: Float,
+  size: Size,
+  density: Density,
+  radius: BpkFlareRadius,
+  pointerDirection: BpkFlarePointerDirection,
+) {
+  val borderRect = when (pointerDirection) {
+    BpkFlarePointerDirection.Up -> Rect(0f, flareHeight, size.width, size.height + RectOffset)
+    BpkFlarePointerDirection.Down -> Rect(0f, -RectOffset, size.width, size.height - flareHeight)
+  }
+  when (radius) {
+    BpkFlareRadius.None -> addRect(borderRect)
+    BpkFlareRadius.Medium -> {
+      val cornerRadius = with(density) { BpkDimension.BorderRadius.Md.toPx() }
+      addRoundRect(RoundRect(borderRect, CornerRadius(cornerRadius, cornerRadius)))
+    }
+  }
+}
+
+private fun Path.addFlare(
+  flareHeight: Float,
+  scale: Float,
+  size: Size,
+  pointerDirection: BpkFlarePointerDirection,
+) {
+  val flareWidth = scale * FlareVectorWidth
+  val flareX = (size.width - flareWidth) / 2
+  val flareY = when (pointerDirection) {
+    BpkFlarePointerDirection.Up -> flareHeight
+    BpkFlarePointerDirection.Down -> size.height - flareHeight
+  }
+  moveTo(flareX + 4.329f * scale, flareY)
+  cubicTo(
+    flareX + 14.663905f * scale, flareY + 0.409248008f * scale,
+    flareX + 24.743092f * scale, flareY + 3.33270635f * scale,
+    flareX + 33.693f * scale, flareY + 8.517f * scale,
+  )
+  lineTo(flareX + 101.736f * scale, flareY + 47.858f * scale)
+  cubicTo(
+    flareX + 112.368f * scale, flareY + 54.043f * scale,
+    flareX + 125.531f * scale, flareY + 54.043f * scale,
+    flareX + 136.264f * scale, flareY + 47.858f * scale,
+  )
+  lineTo(flareX + 204.408f * scale, flareY + 8.518f * scale)
+  cubicTo(
+    flareX + 213.318f * scale, flareY + 3.345f * scale,
+    flareX + 223.396f * scale, flareY + 0.303f * scale,
+    flareX + 233.724f * scale, flareY + 0f * scale,
+  )
+  lineTo(flareX + 238f * scale, flareY + 0f * scale)
+}
+
+private const val FlareVectorHeight = 53
+private const val FlareVectorWidth = 234
+internal val FlareHeight = 11f.dp
```

Comment:
Why internal?

Reply:
it's also used for setting the content padding